### PR TITLE
[5.3] Fix typehint and class names issues

### PR DESCRIPTION
**Pull request by rkgrep**
Yet another regular fixes pack 😄

### CODE REVIEW

File: src/Illuminate/Cache/TaggableStore.php [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 namespace Illuminate\Cache;
 
-abstract class TaggableStore
+use Illuminate\Contracts\Cache\Store;
```
Comment (GrahamCampbell) [inline src/Illuminate/Cache/TaggableStore.php]: what happened here?

File: src/Illuminate/Foundation/helpers.php [hunk 2]
```diff
@@ -199,6 +199,7 @@ function bcrypt($value, $options = [])
      *
      * @param  dynamic  key|key,default|data,expiration|null
      * @return mixed
+     * @throws \Exception
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/helpers.php]: missing newline

File: src/Illuminate/Cache/TaggableStore.php (hunk 1 above)
Comment (rkgrep) [inline src/Illuminate/Cache/TaggableStore.php]: Both `TaggedCache` and `TagSet` constuctors are typehinted with `Store` contract. Which means `TaggableStore` must implement it in order to work properly.
https://github.com/rkgrep/framework/blob/c163e00964207092bd627c7fc6fc46eb1c206c8e/src/Illuminate/Cache/TaggableStore.php#L17

Comment (rkgrep) [inline src/Illuminate/Cache/TaggableStore.php]: Maybe, I should do a separate PR for this

**Comment by taylorotwell** [comment]
Has conflicts.
